kube-core/src/metadata.rs

```diff
@@ -12,3 +14,50 @@ pub struct TypeMeta {
     /// The name of the API
     pub kind: String,
 }
+
+/// Generic representation of any object with ObjectMeta. Allows clients to get
+/// access to a particular ObjectMeta schema without knowing the details of the
+/// version
+#[derive(Deserialize, Serialize, Clone, Default, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct PartialObjectMeta {
```

Comment:
Given that this is actually a wrapper container around a object, I think we should maybe call this `PartialObject`

Reply:
actually, no. let's defer to apimachinery for the root type: https://github.com/kubernetes/apimachinery/blob/7687996c715ee7d5c8cf1e3215e607eb065a4221/pkg/apis/meta/v1/types.go#L1416-L1425 (`PartialObject` implies that is could be deconstructed in other ways, and our `DynamicObject` type is kind of called Unstructured upstream)

oh, and btw, you can even take docstrings from there.

Reply:
Mm yeah that makes sense. I also updated with client-go's docstrings 👍🏻